docker/docker-compose.yml

```diff
@@ -0,0 +1,37 @@
+version: "3.6"
+
+x-omnia: &default
+  image: makerdao/omnia:latest
+  expose:
+    - ${SSB_NET_PORT}
+    - ${SSB_WS_PORT}
+  environment: &env
+    SSB_FILE_SET: /src/ssb-config.json
+    ETH_GAS: 7000000
+    SSB_SET_connections_incoming_net__external: '"${EXT_IP}"'
+    SSB_SET_connections_incoming_net__port: ${SSB_NET_PORT}
+    SSB_SET_connections_incoming_ws__external: '"${EXT_IP}"'
+    SSB_SET_connections_incoming_ws__port: ${SSB_WS_PORT}
+    SSB_SET_caps: ${SSB_CAPS}
+    OMNIA_SET_ethereum_from: '"${ETH_FROM}"'
```

Comment:
Picking, but any chance we can quote this in code? Double quotes aren't too nice and high chance the user will not know that `jq` needs them or forget to put them.

Reply:
I see the `confer` `<PREFIX>_SET_*` variables as a general low level means of modify a config/JSON-file at runtime. To be able to set any type of JSON value (i.e. numbers, strings, objects and arrays) we need to be able to explicitly express this in the variables.

With this in mind I agree that this is an abstraction that might be confusing to an end user. That is why I see the `.env` file as the user facing interface where any variable set there will be properly wrapped in its correct JSON syntax inside the `docker-compose.yml` file.

Reply:
Again, not a huge deal, but depends on the user I guess. For a quick local dev stack you may use `docker-compose`, for a user of the container, she may want to set the vars directly. Eg. this is how we're setting the values currently on the devops side when deploying the container on AWS ECS - https://github.com/makerdao/infrastructure/blob/omnia-ecs/develop/omnia-oracle/env-vars.tf. Notice the extra quotation needed which is not very pretty.

Reply:
If it is just about the double-quotes looking messy we can introduce `OMNIA_SETSTR_*` which wouldn't require double-quotes. But In the end the user would need to understand the abstraction of `jq` and the structure of the Omnia config anyway.

The point was just to be able to manipulate a configuration in a general way without having to rebuild the image or create a custom script that covers each use case.

This said, if we think that the end user interface is the Docker image itself we should move the high-level env vars (declared in `docker-compose.yml`) to a script wrapper baked into the image.

What do you think @eskp ?

Reply:
Yes I think having Docker image as the interface makes sense. We can copy `docker-entrypoint.sh` inside the image, which sets these messy vars and kicks off the runner. Like I had in my PR.

Reply:
Great!

The runner already takes this role though, I would just have it source a `pre-start` script that could set vars.

Reply:
Happy to leave it as nice to have for now. We may need to get back to this in the near future when Surf is included as that will have its own set of vars.

Reply:
A compromise for now is the added feature of setting JSON config strings with `OMNIA_STR_*=some_unquoted_string_here`.